x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/common/http/HttpRequest.java

```diff
@@ -511,7 +512,7 @@ public static InputStream filterToXContent(HttpRequest request, XContentType xCo
         try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
              XContentBuilder filteredBuilder = new XContentBuilder(xContentType.xContent(), bos,
                  Collections.emptySet(), Collections.singleton(excludeField),
-                 xContentType.toParsedMediaType())) {
+                 xContentType.toParsedMediaType(), RestApiVersion.current())) {
```

Comment:
I hate the idea of *more* constructors ... but I wonder if it makes sense here to de-emphasize that RestApiVersion is non-consequential. 

The additional constructor, would be the original constructor prior to the PR that would  set RestAPIVersion to current(). That way use cases like this (and enrich) that don't care about REST API compatibility don't need to concern themselves here.